sql/core/src/test/scala/org/apache/spark/sql/SubquerySuite.scala

```diff
@@ -938,6 +938,46 @@ class SubquerySuite extends QueryTest
     }
   }
 
+  test("SPARK-36124: Correlated subqueries with set operations") {
+    withTempView("t0", "t1", "t2") {
+      Seq((1, 1), (2, 0)).toDF("t0a", "t0b").createOrReplaceTempView("t0")
+      Seq((1, 1, 3)).toDF("t1a", "t1b", "t1c").createOrReplaceTempView("t1")
+      Seq((1, 1, 5), (2, 2, 7)).toDF("t2a", "t2b", "t2c").createOrReplaceTempView("t2")
+
+      // Union with different outer refs
+      checkAnswer(
```

Comment:
I think we already checked the answer in the SQL query test. Here we can check the plan structure if necessary instead of the answer.